tools/cfs-fuse.c

```diff
@@ -973,16 +961,72 @@ static void cfs_release(fuse_req_t req, fuse_ino_t ino, struct fuse_file_info *f
 	fuse_reply_err(req, 0);
 }
 
+static void cfs_read_inline(fuse_req_t req, fuse_ino_t ino, size_t size,
+			    off_t offset, struct fuse_file_info *fi)
+{
+	const erofs_inode *cino = cfs_get_erofs_inode(ino);
+	uint32_t mode;
+	uint64_t file_size;
+	uint16_t xattr_icount;
+	size_t xattr_size;
+	size_t isize;
+	uint64_t n_blocks;
+	uint64_t last_oob_block;
+	bool tailpacked;
+	size_t tail_size;
+	uint32_t raw_blkaddr;
+	off_t oob_size;
+	const uint8_t *tail_data;
+	const uint8_t *oob_data;
+	struct iovec iov[2];
+	int i;
```

Comment:
(IMO, this is crying out for declare-and-initialize...)

Reply:
Not to completely belabor this point but...basically all other sane modern languages allow declaring (and initializing) a variable inside a function.

In the case of this particular function, it would probably be *half the size* if done this way.

I know you work on the Linux kernel which isn't yet using this style, and I know even some others like systemd haven't really adopted it but IMO it really just makes things much more readable.  Particularly in this example!

Reply:
Actually, the kernel is moving toward using __attribute__(cleanup), and as a result will be moving towards this style too (at least partially)

Reply:
https://lwn.net/ml/linux-kernel/CAHk-=wi-RyoUhbChiVaJZoZXheAwnJ7OO=Gxe85BkPAd93TwDA@mail.gmail.com/